### PR TITLE
Implements durable PUTs in Python client

### PR DESCRIPTION
**Pull request by vsreekanti**
I haven't touched the C++ CLI yet, but it's not being used in anything. We should convert that into a real client library and implement the same thing there, but the Python client should be sufficient for now.

### CODE REVIEW

File: kvs/src/client/python/anna/client.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+            req_ids.append(req.request_id)
+
+        responses = recv_response(req_ids, self.response_puller, KeyResponse)
+
+        for resp in responses:
+            tup = resp.tuples[0]
+            if tup.invalidate:
+                # reissue the request
+                return self.durable_put(key, value)
```
Comment (cw75) [inline kvs/src/client/python/anna/client.py]: where are we invalidating the cache? I don't see a call to `_invalidate_cache`

File: kvs/src/client/python/anna/client.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+            if tup.error != 0:
```
Comment (cw75) [inline kvs/src/client/python/anna/client.py]: Just to confirm: if invalidation is false, then in principle `tup.error` should be 0 right? So if `tup.error != 0` something must be wrong so we are just returning false?

File: kvs/src/client/python/anna/client.py (hunk 1 above)
Comment (vsreekanti) [reply kvs/src/client/python/anna/client.py]: Ah, good catch. Will fix.

File: kvs/src/client/python/anna/client.py (hunk 2 above)
Comment (vsreekanti) [reply kvs/src/client/python/anna/client.py]: Yes, `False` just means that the operation failed, and you should try again.

**Pull request merged**